src/d3a/models/area/event_dispatcher.py

```diff
@@ -79,6 +79,7 @@ def _broadcast_notification(self, event_type: Union[MarketEvent, AreaEvent], **k
         # Broadcast to children in random order to ensure fairness
         for child in sorted(self.area.children, key=lambda _: random()):
             child.dispatcher.event_listener(event_type, **kwargs)
+        kwargs.pop('current_tick', None)
```

Comment:
I understand the need to pass the current_tick when broadcasting activate, so that the child areas can immediately sync with their root parent area. However, does it really hurt to remove it from here ? I think it would be fine to leave it as it is.

Reply:
https://github.com/gridsingularity/d3a/pull/1021#discussion_r562698227